### ISSUE TITLE
Add Plant Disease Detection Module for Crop Health Monitoring

### ISSUE THREAD

**Issue by Vidulaa03** [Eshajha19/agri#230]
### Description
Introduce a new feature that allows users to upload an image of a plant leaf and detect possible diseases affecting the crop.

This enhancement can help farmers identify common plant diseases early and take preventive measures to reduce crop loss.

#### Frontend
- Add a new **Plant Disease Detection** section/page
- Image upload input (drag & drop or browse)
- Image preview before submission
- Analyze button
- Loading state while processing
- Result card displaying:
  - Disease Name
  - Confidence Score
  - Recommended Treatment / Suggestions

#### Backend
- Add API endpoint for disease prediction (mock or actual model integration)
- Accept uploaded image
- Return structured JSON response

#### Example API Response
```json
{
  "disease": "Powdery Mildew",
  "confidence": "94%",
  "solution": "Apply sulfur-based fungicide and avoid overhead watering."
}
```
### Why This Matters
- Helps farmers detect diseases early  
- Increases platform usefulness  
- Adds AI/ML-ready functionality  
- Strong real-world agriculture use case  

### Additional Notes
Initial implementation can use mock responses, with future scope for TensorFlow / PyTorch model integration.

I'd be happy to work on this under NSOC'26 if assigned!

**Comment by Eshajha19** [comment]
Assigned @Vidulaa03

**Comment by SujalTripathi** [comment]
hey @Eshajha19 there is almost 2 weeks so can i work on these issue and can you please change the label to level3 bcz required to work on both fronted and backend with drag drop functionltiy implementation

**Comment by Eshajha19** [comment]
> hey [@Eshajha19](https://github.com/Eshajha19) there is almost 2 weeks so can i work on these issue and can you please change the label to level3 bcz required to work on both fronted and backend with drag drop functionltiy implementation

Hi @SujalTripathi the crop Disease detection module is already created if there are any improvements then you can raise an issue for that or provide an implementation plan here.
I will assign the issue and change the Labels also.

**Comment by SujalTripathi** [comment]
okay @Eshajha19 i will let you know if any does. can you please review my open PR

**Comment by Eshajha19** [comment]
> okay [@Eshajha19](https://github.com/Eshajha19) i will let you know if any does. can you please review my open PR

Sure I will review it once deployment errors are fixed.
